safehttp/plugins/csp/csp.go

```diff
@@ -159,6 +162,15 @@ func (f FramingPolicy) Serialize(nonce string) string {
 	var b strings.Builder
 	b.WriteString("frame-ancestors 'self'")
```

Comment:
At this point I'd carve out the `frame-ancestors` and `report-uri` directives to separate functions and join them with `; `.

e.g.:

```
ancestors := frameAncestors(f.Sources)
rep := report(f.ReportURI)

return strings.Join([]string{ancestors, rep}, "; ")
```

Alternatively:

```
var b strings.Builder
f.writeAncestors(&b)
f.writeReportURI(&b)
return b.String()
```

(the `writeFoo` methods always append `; `)

Reply:
I chose the first option. It adds a semicolon to the default policy but it's still valid.